tests/integration/conftest.py

```diff
@@ -1093,6 +1107,10 @@ def create_transactions_event_table_from_data_source(
             "ÀMOUNT": "FLOAT",
             "TZ_OFFSET": "VARCHAR",
             "TRANSACTION_ID": "VARCHAR",
+            "EMBEDDING_ARRAY": "ARRAY",
+            "ARRAY": "ARRAY",
+            "FLAT_DICT": "OBJECT" if data_source.type == SourceType.SNOWFLAKE else "STRUCT",
+            "NESTED_DICT": "OBJECT" if data_source.type == SourceType.SNOWFLAKE else "STRUCT",
```

Comment:
🤔 Should we introduce `OBJECT` type here or reuse existing `STRUCT` type?

Reply:
Snowflake doesn't support STRUCT and treats dicts as OBJECT.  
Spark treats dicts and STRUCT. 
How would we reuse only one? 

We do same for `TIMESTAMP_TZ` don't we? 
https://github.com/featurebyte/featurebyte/blob/vikua/DEV-2540-service/tests/integration/conftest.py#L1099-L1100

Reply:
This is the part i am not sure. If `STRUCT` and `OBJECT` have different characteristics (other than due to different feature store type), it makes sense to have two different dtypes. Just want to check whether we should keep one or having two different types. For the `TIMESTAMP` & `TIMESTAMP_TZ`, there are different as one of them have timezone info whereas another one does not have. 

@YS-L Would be good if you can confirm this part.

Reply:
I agree it should probably be something like DICT type for everything to abstract away from DWH details. 
But I'm a bit hesitant to add mapping for STRUCT and OBJECT to new DICT type in this PR. 
Not sure how big the blast radius can be.

Reply:
Got it, I'm fine with the current implementation. Would be good to revisit and clean up if possible.

Reply:
https://featurebyte.atlassian.net/browse/DEV-2612 to track